src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -657,7 +657,7 @@ public void CopyTo(Array array, int index)
                     Debug.Assert(span.Length > 0);
                     Debug.Assert(m_array.Length > quotient);
                     // mask the final byte
-                    span[span.Length - 1] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
+                    span[remainder] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
```

Comment:
Before the change from https://github.com/dotnet/corefx/pull/33367, this used to be as follows:
https://github.com/dotnet/corefx/blob/8716f1adc2f583d54b825998838e7cf7ed64cc0f/src/System.Collections/src/System/Collections/BitArray.cs#L545-L554

Before, we were writing one **byte** at a time, and then if the bit array contained extra **bits** to be copied over, we used the next index to write those bits. Now, we are writing **4 bytes** at a time, and if we have 1-3 bytes left to write, we have the switch on remainder below. Beyond that, any extra bits left to copy (i.e. < 1 byte) should go at offset `remainder`.

I believe using `remainder` as the offset to write the last byte to is correct (and the test pass). Please review/verify.

Reply:
By the way
```csharp
(byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
```
can be optimized in JIT: https://godbolt.org/z/XQ9Mpe
current codegen: [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aaGeHgBPDDAZDcGABQBLAHYYG2NAxlzY+bDIAmMKIuUMYCDFGwAhSRlwBKDt14BIYgHYG4gUMvjx2BgD4fL1XVpLSgGACoGAA5LawAyF3FyBgAeZJdlSwMjU3MrBjgGchiAblseAF8acqA===)

Reply:
cc @AndyAyersMS